src/runtime_src/core/pcie/driver/linux/xocl/xocl_fdt.c

```diff
@@ -1461,6 +1461,87 @@ static void xocl_pack_subdev(xdev_handle_t xdev_hdl, struct xocl_subdev *subdev)
 			i * XOCL_SUBDEV_RES_NAME_LEN;
 }
 
+static int xocl_fdt_get_pci_addr(xdev_handle_t xdev_hdl)
+{
+        struct xocl_dev_core    *core = XDEV(xdev_hdl);
+        int offset;
+        const u32* bar;
+        const u32* pfn;
+        const u64* base_addr;
+        const u64* range;
+        u32 pf, bar_idx;
+        int ret = 0, count = 0;
+
+#if PF == MGMTPF
+        pf = 0;
+#else
+        pf = 1;
+#endif
+
+        if (!core->fdt_blob) {
+                xocl_xdev_err(xdev_hdl, "fdt blob is empty");
+                return -EINVAL;
+        }
+
+        offset = fdt_path_offset(core->fdt_blob, "/pcie/bars");
+        if (offset < 0) {
+                xocl_xdev_dbg(xdev_hdl, "pcie bars nodes are not present in firmware data");
+                return -EINVAL;
+        }
+
+        core->bars = kmalloc(sizeof(struct pci_bars) * NUM_PCI_BARS, GFP_KERNEL);
+        if (!core->bars) {
+                ret = -ENOMEM;
+                goto done;
+        }
+        memset(core->bars, 0, sizeof(struct pci_bars) * NUM_PCI_BARS);
+
+        for (offset = fdt_first_subnode(core->fdt_blob, offset);
+                offset >= 0;
+                offset = fdt_next_subnode(core->fdt_blob, offset)) {
+
+                pfn = fdt_getprop(core->fdt_blob, offset,"physical_function", NULL);
```

Comment:
please define a macro for "physical_function" in xocl_fdt.h. Actually, could we consider to change the xclbinutil tool to generate unified keywords which is using by endpoints. E.g.

#define PROP_PF_NUM "pcie_physical_function"
#define PROP_BAR_IDX "pcie_bar_mapping"
#define PROP_IO_OFFSET "reg"

Reply:
Discussing with @rozumx .